pyccl/halos/profiles_2pt.py

```diff
@@ -71,14 +73,25 @@ def fourier_2pt(self, cosmo, k, M, a, prof, *, prof2=None):
         if prof2 is None:
             prof2 = prof
 
+        HP = HaloProfile
+        if not (isinstance(prof, HP) and isinstance(prof2, HP)):
+            raise TypeError("prof and prof2 must be HaloProfile")
```

Comment:
Do we need this?

Reply:
I don't remember why I added it. I think I added it because I was having some errors while testing my merge with master, probably because the order of the arguments changed and I was passing the wrong object. I can remove it if you think it will not be helpful.

Reply:
meh, let's leave it.